bookkeeper-server/src/main/java/org/apache/bookkeeper/verifier/BookkeeperVerifier.java

```diff
@@ -0,0 +1,692 @@
+    private final int targetConcurrentReads;
+    private final double coldToHotRatio;
+
+    private final long targetLedgerEntries;
+
+    BookkeeperVerifier(
+            BookkeeperDriver driver,
+            int ensembleSize,
+            int writeQuorum,
+            int ackQuorum,
+            int duration,
+            int drainTimeout,
+            int targetConcurrentLedgers,
+            int targetConcurrentWrites,
+            int targetWriteGroup,
+            int targetReadGroup,
+            int targetLedgers,
+            long targetLedgerSize,
+            int targetEntrySize,
+            int targetConcurrentReads,
+            double coldToHotRatio) {
+        this.driver = driver;
+        this.ensembleSize = ensembleSize;
+        this.writeQuorum = writeQuorum;
+        this.ackQuorum = ackQuorum;
+        this.duration = duration;
+        this.drainTimeout = drainTimeout;
+        this.targetConcurrentLedgers = targetConcurrentLedgers;
+        this.targetConcurrentWrites = targetConcurrentWrites;
+        this.targetWriteGroup = targetWriteGroup;
+        this.targetReadGroup = targetReadGroup;
+        this.targetLedgers = targetLedgers;
+        this.targetEntrySize = targetEntrySize;
+        this.targetConcurrentReads = targetConcurrentReads;
+        this.coldToHotRatio = coldToHotRatio;
+
+        this.targetLedgerEntries = targetLedgerSize / targetEntrySize;
+    }
+
+    private int outstandingWriteCount = 0;
+    private int outstandingReadCount = 0;
+    private long nextLedger = 0;
+    private long getNextLedgerID() {
+        return nextLedger++;
+    }
+
+    /**
+     * State required to regenerate an entry.
+     */
+    class EntryInfo {
+        private long entryID;
+        private long seed;
+        EntryInfo(long entryID, long seed) {
+            this.entryID = entryID;
+            this.seed = seed;
+        }
+        byte[] getBuffer() {
+            Random rand = new Random(seed);
+            byte[] ret = new byte[targetEntrySize];
+            rand.nextBytes(ret);
+            return ret;
+        }
+        long getEntryID() {
+            return entryID;
+        }
+    }
+
+    /**
+     * Contains the state required to reconstruct the contents of any entry in the ledger.
+     * The seed value passed into the constructor fully determines the contents of the
+     * ledger.  Each EntryInfo has its own seed generated sequentially from a Random instance
+     * seeded from the original seed.  It then uses that seed to generate a secondary Random
+     * instance for generating the bytes within the entry.  See EntryIterator for details.
+     * Random(seed)
+     *  |
+     *  E0 -> Random(E0) -> getBuffer()
+     *  |
+     *  E1 -> Random(E1) -> getBuffer()
+     *  |
+     *  E2 -> Random(E2) -> getBuffer()
+     *  |
+     *  E3 -> Random(E3) -> getBuffer()
+     *  |
+     *  E4 -> Random(E4) -> getBuffer()
+     *  |
+     *  ...
+     */
+    class LedgerInfo {
```

Comment:
static?

Reply:
It allocates EntryInfo's, easier to leave it non-static.